src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -278,7 +281,12 @@ private static async Task<ExtensionMethodImportCompletionCacheEntry> GetUpToDate
             }
 
             cacheEntry = builder.ToCacheEntry();
-            cacheService.ProjectItemsCache[project.Id] = cacheEntry;
+#if NET
+            s_projectItemsCache.AddOrUpdate(project.State, cacheEntry);
+#else
+            s_projectItemsCache.Remove(project.State);
+            s_projectItemsCache.GetValue(project.State, _ => cacheEntry);
```

Comment:
Should the result of this be put into cacheEntry?

Reply:
it should be fine either way.  i prefer this way so that we ensure that we exactly return the data we computed for this exact arguments.